plugins/outputs/kafka/kafka.go

```diff
@@ -377,13 +404,26 @@ func (k *Kafka) Write(metrics []telegraf.Metric) error {
 					k.Log.Error("The timestamp of the message is out of acceptable range, consider increasing broker `message.timestamp.difference.max.ms`; dropping batch")
 					return nil
 				}
-				return prodErr
+				unsentMsgs[i] = prodErr.Msg
```

Comment:
does this need to be copied or is it okay to re-use? Not sure what the ownership of the message would be for the error cases - I suppose this code owns this object, correct?

Reply:
We are the creators of this object that we are reusing here. I also don't know think there is a way for them to destroy the object (or at least it would be bad practice). I think reusing is fine